search/feature_offset_match.hpp

```diff
@@ -196,6 +196,14 @@ class TrieValuesHolder
 template <typename DFA>
 struct SearchTrieRequest
 {
+  SearchTrieRequest() = default;
+
+  SearchTrieRequest(SearchTrieRequest const &) = delete;
+  SearchTrieRequest & operator=(SearchTrieRequest const &) = delete;
+
+  SearchTrieRequest(SearchTrieRequest &&) = default;
+  SearchTrieRequest & operator=(SearchTrieRequest &&) = default;
```

Comment:
- Is it needed to explicitly declare ctors and _moves_?
- Put noexcept

Reply:
Move constructors are needed to make MSVC happy:
```
  FAILED: search/CMakeFiles/search.dir/geocoder.cpp.obj 
C:\Program Files\Microsoft Visual Studio\2022\Preview\VC\Tools\MSVC\14.40.33617\include\xmemory(748): error C2280: 'strings::LevenshteinDFA::LevenshteinDFA(const strings::LevenshteinDFA &)': attempting to reference a deleted function
  base/levenshtein_dfa.hpp(97): note: see declaration of 'strings::LevenshteinDFA::LevenshteinDFA'
  base/levenshtein_dfa.hpp(97): note: 'strings::LevenshteinDFA::LevenshteinDFA(const strings::LevenshteinDFA &)': function was explicitly deleted
  C:\Program Files\Microsoft Visual Studio\2022\Preview\VC\Tools\MSVC\14.40.33617\include\xmemory(748): note: the template instantiation context (the oldest one first) is
  search/ranking_utils.hpp(190): note: see reference to class template instantiation 'std::vector<strings::LevenshteinDFA,std::allocator<strings::LevenshteinDFA>>' being compiled
  C:\Program Files\Microsoft Visual Studio\2022\Preview\VC\Tools\MSVC\14.40.33617\include\vector(676): note: while compiling class template member function 'std::vector<strings::LevenshteinDFA,std::allocator<strings::LevenshteinDFA>>::vector(const std::vector<strings::LevenshteinDFA,std::allocator<strings::LevenshteinDFA>> &)'
  search/feature_offset_match.hpp(224): note: see the first reference to 'std::vector<strings::LevenshteinDFA,std::allocator<strings::LevenshteinDFA>>::vector' in 'search::SearchTrieRequest<strings::LevenshteinDFA>::SearchTrieRequest'
  C:\Program Files\Microsoft Visual Studio\2022\Preview\VC\Tools\MSVC\14.40.33617\include\vector(680): note: see reference to function template instantiation 'void std::vector<strings::LevenshteinDFA,std::allocator<strings::LevenshteinDFA>>::_Construct_n<strings::LevenshteinDFA*const &,strings::LevenshteinDFA*const &>(const unsigned __int64,strings::LevenshteinDFA *const &,strings::LevenshteinDFA *const &)' being compiled
  C:\Program Files\Microsoft Visual Studio\2022\Preview\VC\Tools\MSVC\14.40.33617\include\vector(2078): note: see reference to function template instantiation 'strings::LevenshteinDFA *std::_Uninitialized_copy<strings::LevenshteinDFA*,strings::LevenshteinDFA*,std::allocator<strings::LevenshteinDFA>>(_InIt,_Se,strings::LevenshteinDFA *,_Alloc &)' being compiled
          with
          [
              _InIt=strings::LevenshteinDFA *,
              _Se=strings::LevenshteinDFA *,
              _Alloc=std::allocator<strings::LevenshteinDFA>
          ]
  C:\Program Files\Microsoft Visual Studio\2022\Preview\VC\Tools\MSVC\14.40.33617\include\xmemory(1882): note: see reference to function template instantiation 'void std::_Uninitialized_backout_al<std::allocator<strings::LevenshteinDFA>>::_Emplace_back<strings::LevenshteinDFA&>(strings::LevenshteinDFA &)' being compiled
  C:\Program Files\Microsoft Visual Studio\2022\Preview\VC\Tools\MSVC\14.40.33617\include\xmemory(1828): note: see reference to function template instantiation 'void std::_Default_allocator_traits<_Alloc>::construct<_Ty,strings::LevenshteinDFA&>(_Alloc &,_Objty *const ,strings::LevenshteinDFA &)' being compiled
          with
          [
              _Alloc=std::allocator<strings::LevenshteinDFA>,
              _Ty=strings::LevenshteinDFA,
              _Objty=strings::LevenshteinDFA
          ]
```

> Put noexcept

Nope. `std::is_nothrow_move_assignable_v<strings::UniStringDFA>` is false and `std::unordered_set` does not have noexcept move constructor.

Reply:
Well, yes, because we should make buffer_vector noexcept first in other PR :)